nightlies,logictestccl: probabilistically run backup+restore during logictests

adityamaru opened the pull request:
This change introduces a nightly script that run the logictests
in logictestccl to run with `COCKROACH_LOGIC_TEST_BACKUP_RESTORE_PROBABILITY`
set to a non-zero value. This environment variable controls the probability of us
running a backup+restore between lines of a logic test. Egs: 1.0 means that
every line of the logic test will be run after a cluster backup and restore.

The motivation for this change is to increase the coverage of backup+restore
testing by piggybacking on our large logictest corpus. To begin with we only
run tests in logictestccl with config local, but this will be gradually expanded
as we work through the failure modes.

Informs: https://github.com/cockroachdb/cockroach/issues/77130

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/78597)
<!-- Reviewable:end -->

adityamaru committed:
nightlies,logictestccl: probabilistically run backup+restore during logic tests

This change introduces a nightly script that run the logictests
in logictestccl to run with `COCKROACH_LOGIC_TEST_BACKUP_RESTORE_PROBABILITY`
set to a non-zero value. This environment variable controls the probability of us
running a backup+restore between lines of a logic test. Egs: 1.0 means that
every line of the logic test will be run after a cluster backup and restore.

The motivation for this change is to increase the coverage of backup+restore
testing by piggybacking on our large logictest corpus. To begin with we only
run tests in logictestccl with config local, but this will be gradually expanded
as we work through the failure modes.

Informs: #77130

Release note: None

adityamaru commented:
This is an example of the nightly run https://teamcity.cockroachdb.com/viewLog.html?buildId=6458217&buildTypeId=Cockroach_Nightlies_SqlLogicBackupRestore&tab=buildResultsDiv&branch_Cockroach_Nightlies=78597, the one failure seems like a legitimate bug which makes me happy - https://github.com/cockroachdb/cockroach/issues/87915#issuecomment-1245618448

adityamaru commented:
Closing in favour of #87918
